internal/ls/findallreferences.go

```diff
@@ -0,0 +1,1490 @@
+	if !isForRenameWithPrefixAndSuffixText(options) || len(symbol.Declarations) == 0 {
+		exportSpecifier = core.Find(symbol.Declarations, ast.IsExportSpecifier)
+	}
+	if exportSpecifier != nil {
+		// !!! not implemented
+
+		// When renaming at an export specifier, rename the export and not the thing being exported.
+		// state.getReferencesAtExportSpecifier(exportSpecifier.Name(), symbol, exportSpecifier.AsExportSpecifier(), state.createSearch(node, originalSymbol, comingFromUnknown /*comingFrom*/, "", nil), true /*addReferencesHere*/, true /*alwaysGetReferences*/)
+	} else if node != nil && node.Kind == ast.KindDefaultKeyword && symbol.Name == ast.InternalSymbolNameDefault && symbol.Parent != nil {
+		state.addReference(node, symbol, entryKindNone)
+		// !!! not implemented
+		// state.searchForImportsOfExport(node, symbol, &ExportInfo{exportingModuleSymbol: symbol.Parent, exportKind: ExportKindDefault})
+	} else {
+		search := state.createSearch(node, symbol, comingFromUnknown /*comingFrom*/, "", state.populateSearchSymbolSet(symbol, node, options.use == referenceUseRename, !!options.providePrefixAndSuffixTextForRename, !!options.implementations))
+		state.getReferencesInContainerOrFiles(symbol, search)
+	}
+
+	return state.result
+}
+
+type ExportKind int
+
+const (
+	ExportKindDefault      ExportKind = 0
+	ExportKindNamed        ExportKind = 1
+	ExportKindExportEquals ExportKind = 2
+)
+
+type ExportInfo struct {
+	exportingModuleSymbol *ast.Symbol
+	exportKind            ExportKind
+}
+
+type comingFromType int
+
+const (
+	comingFromUnknown comingFromType = 0
+	comingFromImport  comingFromType = 1
+	comingFromExport  comingFromType = 2
+)
+
+/**
+* Symbol that is currently being searched for.
+* This will be replaced if we find an alias for the symbol.
+ */
+type refSearch struct {
+	/** If coming from an export, we will not recursively search for the imported symbol (since that's where we came from). */
+	comingFrom comingFromType // import, export
+
+	symbol      *ast.Symbol
+	text        string
+	escapedText string
+	/** Only set if `options.implementations` is true. These are the symbols checked to get the implementations of a property access. */
+	parents []*ast.Symbol
+
+	allSearchSymbols []*ast.Symbol
+
+	/**
+	* Whether a symbol is in the search set.
+	* Do not compare directly to `symbol` because there may be related symbols to search for. See `populateSearchSymbolSet`.
+	 */
+	includes func(symbol *ast.Symbol) bool
+}
+
+// type (
+// 	ImportTracker = func(exportSymbol *ast.Symbol, exportInfo ExportInfo, isForRename bool) ImportsResult
+// 	ImportsResult struct {
+// 		importSearches []struct {
+// 			importLocation *ast.ModuleExportName
+// 			importSymbol   *ast.Symbol
+// 		}
+// 		singleReferences []*ast.Node // ientifier | stringliteral
+// 		indirectUsers    []*ast.SourceFile
+// 	}
+// )
+
+type seenTracker[T comparable] struct {
+	m core.Set[T]
+}
+
+func (t *seenTracker[T]) mark(key T) bool {
```

Comment:
I think you could just get rid of the new struct and use `core.Set`, and add this as a set method. I recently added method `AddIfAbsent` to replace Strada's `addToSeen`.